packages/@sanity/migrate/src/it-utils/lastValueFrom.ts

```diff
@@ -0,0 +1,27 @@
+interface Options<T> {
+  defaultValue?: T
+}
+export async function lastValueFrom<T>(
+  it: AsyncIterableIterator<T>,
+  options?: Options<T>,
+): Promise<T> {
+  const defaultGiven = 'defaultValue' in (options ?? {})
+  let latestValue: T | undefined
+  let didYield = false
+
+  for await (const value of it) {
+    didYield = true
+    latestValue = value
+  }
+  if (!didYield) {
+    if (defaultGiven) {
+      return options!.defaultValue!
+    }
+    if (!defaultGiven && !didYield) {
```

Comment:
I may be mistaken, but isn't this entire nested `if (!defaultGiven && !didYield)` statement redundant?

Reply:
hah, yes - good catch, thanks!